master/buildbot/data/resultspec.py

```diff
@@ -49,17 +49,26 @@ class FieldBase(object):
         'contains': lambda d, v: set(v) <= set(d),
```

Comment:
looks like this 'old' code should provide a OR behaviour at data API level. 
It is just the sql mode which was not implemented